dali/operators/generic/resize/tensor_resize.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_RESIZE_TENSOR_RESIZE_H_
+#define DALI_OPERATORS_GENERIC_RESIZE_TENSOR_RESIZE_H_
+
+#include <cassert>
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/operators/image/resize/resize_base.h"
+#include "dali/operators/image/resize/tensor_resize_attr.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace tensor_resize {
+
+
+template <typename Backend>
+class TensorResize : public Operator<Backend>
+                   , protected ResizeBase<Backend> {
+ public:
+  explicit TensorResize(const OpSpec &spec);
+
+ protected:
+  int NumSpatialDims() const { return resize_attr_.spatial_ndim_; }
+  int FirstSpatialDim() const { return resize_attr_.first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+
+  void RunImpl(Workspace &ws) override;
+
+  void PrepareParams(const ArgumentWorkspace &ws, const TensorListShape<> &input_shape) {
+    int nsamples = input_shape.num_samples();
+    resize_attr_.PrepareResizeParams(spec_, ws, input_shape);
+
+    if (NumSpatialDims() < 1) {
+      throw std::invalid_argument(
+          "TensorResize expects at least one spatial dimension to be resized");
+    }
+
```

Comment:
Perhaps for the time being, you can simply report 1D as not implemented and keep the operator in `experimental`.
```suggestion

    if (NumSpatialDims() == 1) {
      throw std::invalid_argument("1D resizing is not implemented.");
    }

```

Reply:
Done